### ISSUE TITLE
Migrate from React.PropTypes to prop-types

### ISSUE THREAD

**Issue by folivi** [mlaursen/react-md#325]
Hi, I'm gettting this errors

Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.

Warning: Motion: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.

How can I fix them?

**Comment by sniphpet** [comment]
Hi,

I think this is due to the next react upgrade. React.PropTypes will be deprecated and replaced by a standalone package [prop-types](https://www.npmjs.com/package/prop-types).

You can find the deprecation notice on the [react documentation page](https://facebook.github.io/react/docs/typechecking-with-proptypes.html).

![Deprecation notice](https://cloud.githubusercontent.com/assets/5574267/24841098/0e6f87fa-1d7c-11e7-80f6-2b351b2b9623.png)

To sum up 
```
import { PropTypes } from 'react';
//need to be replaced by
import PropTypes  from 'prop-types';

```

**Comment by folivi** [comment]
hi,
from what I see in devtools console, the errors seem to come from Card.js and Motion.js in the md package

**Comment by leedstyh** [comment]
We can simply wait for @mlaursen to fix it.

The warning is not serious right now, your app will work fine.

**Comment by folivi** [comment]
@leedstyh just spent some time playing with the components. They seem to be working fine

**Comment by mlaursen** [comment]
Yeah, I'm a little unsure about how to handle this migration. I'll need to wait for [react-motion to update](https://github.com/chenglou/react-motion/issues/448) and also figure out what the new convention for the `prop-types` is. Is this another `peerDependency` like `react`/`react-dom`, or is it more static so I can just use it as a local `devDependency`?

**Comment by joseds** [comment]
Hi, when will these changed be merged with the main branch? Currently my Jest snapshot tests are not working because of this. 

Please take your time, I don't mean to be impatient, it's just for planning ahead. :smile:

**Comment by mlaursen** [comment]
Right now, I am unable to merge due to [this bug](https://github.com/reactjs/react-transition-group/issues/29) in the new react-transition-group. Once it has been fixed, I'll do a patch with this changes as well. So hopefully really soon!

**Comment by joseds** [comment]
Thanks for the information!
